src/ui/ServerAdministration/index.tsx

```diff
@@ -181,17 +183,29 @@ class ServerAdministrationContainer
 
   // TODO: Once the user serializes better, this method should be moved to the ./realms/RealmsTableContainer.tsx
   private onRealmOpened = async (path: string) => {
-    if (!this.state.isRealmOpening) {
+    if (!this.state.isRealmOpening && this.adminRealm) {
       this.setState({ isRealmOpening: true });
       try {
-        // Let the UI update before sync waiting on the window to appear
         const realm: ISyncedRealmToLoad = {
           user: this.props.user,
           mode: RealmLoadingMode.Synced,
           path,
           validateCertificates: this.props.validateCertificates,
         };
-        await main.showRealmBrowser({ realm });
+        const realmFile = this.adminRealm.objectForPrimaryKey<ros.RealmFile>(
+          'RealmFile',
+          path,
+        );
+        if (realmFile) {
+          if (realmFile.realmType === 'partial') {
+            await this.showPartialRealmWarning();
+            await main.showRealmBrowser({ realm, readOnly: true });
+          } else {
+            await main.showRealmBrowser({ realm });
+          }
+        } else {
+          throw new Error("Couldn't find the Realm in the /__admin Realm");
```

Comment:
perhaps mention which realm? Or is that obvious from the context?

Reply:
You just clicked open on a specific Realm, so yeah - I would say so.